### ISSUE TITLE
update macro statement error handling to provide helpful messaging to user

### ISSUE THREAD

**Issue by hamilton** [jwills/dbt-duckdb#5]
Howdy partner,

I ran into an error creating a minimal dbt-duckdb example. I wish I could give more information but I'm a dbt newbie. Below is a repository that (hopefully) recreates this issue w/ a couple of essentially trivial model files:

https://github.com/hamilton/dbt-duckdb-test

`dbt run` gives this output:

```
00:08:05  Running with dbt=1.0.1
00:08:05  Found 2 models, 4 tests, 0 snapshots, 0 analyses, 165 macros, 0 operations, 0 seed files, 0 sources, 0 exposures, 0 metrics
00:08:05  
00:08:05  Concurrency: 1 threads (target='dev')
00:08:05  
00:08:05  1 of 2 START table model main.ingest............................................ [RUN]
00:08:05  1 of 2 ERROR creating table model main.ingest................................... [ERROR in 0.08s]
00:08:05  2 of 2 SKIP relation main.second................................................ [SKIP]
00:08:05  
00:08:05  Finished running 1 table model, 1 view model in 0.34s.
00:08:05  
00:08:05  Completed with 1 error and 0 warnings:
00:08:05  
00:08:05  Compilation Error in macro statement (macros/etc/statement.sql)
00:08:05    cannot unpack non-iterable NoneType object
00:08:05    
00:08:05    > in macro materialization_table_default (macros/materializations/models/table/table.sql)
00:08:05    > called by macro statement (macros/etc/statement.sql)
00:08:05  
00:08:05  Done. PASS=0 WARN=0 ERROR=1 SKIP=1 TOTAL=2
```

My `profiles.yml`:

```yaml
duckdb_test:
  outputs:
    dev:
      path: /Users/hamiltonulmer/Code/tmp/duckdb_test/db.duckdb
      type: duckdb
  target: dev
```

**Comment by jwills** [comment]
Thank you @hamilton ! pretty sure this is some API update I missed when I did the upgrade, will dig into it and figure it out

**Comment by jwills** [comment]
Hey @hamilton-- this took me a stupid long time to track down, but the issue is in model definitions themselves, which should not have semi-colons at the end of the files for `ingest.sql` and `second.sql`; if you remove them everything works as expected. However, this is an obvious case where I need a much better error msg that I'm getting right now, so I read some other adapter impls and got some ideas on how I can make these errors more obvious to the user. Thank you so much!

**Comment by hamilton** [comment]
Yikes! Thanks for putting in time to figure this out, and sorry it was a newbie error. Will update the title to reflect this.

**Comment by jwills** [comment]
Fix is in 1.0.2

**Comment by hamilton** [comment]
what a mensch! thanks for the fix.
